tests/mail/tests.py

```diff
@@ -569,7 +569,10 @@ def test_backend_arg(self):
                 filebased.EmailBackend
             )
 
-        msg = 'expected str, bytes or os.PathLike object, not object'
+        if sys.platform == 'win32':
+            msg = '_getfullpathname: path should be string, bytes or os.PathLike, not object'
+        else:
+            msg = 'expected str, bytes or os.PathLike object, not object'
```

Comment:
TBH I'm not sure if we need to check message in this case. It's not our message, and this can cause other (OS/versions) adjustments in the future :thinking: .

Reply:
I understand the concern with a changing message.

I normally like checking the message as it demonstrates that the error that occurred is actually the one we expect, and not some other unexpected `TypeError` in some far off code.

Due to the platform specific concerns, I can see going either way here. So let me know if that is preferred and I'll drop it.

Reply:
\cc @carltongibson

Reply:
OK, let's not thrash too long here... 

* I bet these messages never change. (If they do, well...)
* **An** alternative might be to use `assertWarnsRegex()`, with a comment:

        # Allowing for Windows:
        with self.assertRaisesRegex(TypeError, r'str(ing)?, bytes or os.PathLike( object)?, not object'):
            mail.get_connection('django.core.mail.backends.filebased.EmailBackend', file_path=object())

  I'm not sure that's any prettier. 

Let's leave it as it is. Thanks both.